### PR TITLE
Minimize JArrayHelper methods `toInteger`, `pivot`, `arrayUnique`

### PR DESCRIPTION
**Pull request by izharaazmi**
Minimize JArrayHelper methods `toInteger`, `pivot`, `arrayUnique` by using `Joomla\Utilities\ArrayHelper` internally. 
Leaving (reverting from #7782) other four methods as it was originally for b/c reasons as mentioned in #8455.

Thanks @rdeutz

### CODE REVIEW

**Comment by izharaazmi** [comment]
@rdeutz I am not sure if this PR makes any sense after the merge of #8455. Please advise. Do I need to PR afresh for `toInteger`, `pivot` and `arrayUnique`?

**Comment by rdeutz** [comment]
@izharaazmi sure it makes sense #8455 is a revert of #7782, we revert the full PR but there where some good changes in it (#7782) and we should merge them.

**Comment by izharaazmi** [comment]
@rdeutz But In the changes (diff) I see it is not comparing to latest staging branch content, i.e. after the revert I though diff should show up new differences. Am I missing something?

**Comment by izharaazmi** [comment]
Oh I just updated my knowledge and I understand how it works, I was just thinking the wrong way :)
Thanks @rdeutz!

**Comment by rdeutz** [comment]
@izharaazmi you're welcome!

**Comment by abuechert** [comment]
I have tested this item :white_check_mark: successfully on 3a8878ee9b5b998d6299662a243611b8140eee82<br /><br />tested @icampus pizza, bugs & fun.
used the functions <code>toInteger()</code>, <code>pivot()</code> and <code>arrayUnique()</code> from <code>JArrayHelper</code> without patch and with patch and compared the outputs. Outputs were identically.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8479">issues.joomla.org/joomla-cms/8479</a>.</sub>

**Comment by mehmetalipamukci** [comment]
I have tested this item :white_check_mark: successfully on 3a8878ee9b5b998d6299662a243611b8140eee82<br /><br />tested @icampus:
besides that JArrayHelper is deprecated, the patch does what it says (Compared class before and after applied patch)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8479">issues.joomla.org/joomla-cms/8479</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8479">issues.joomla.org/joomla-cms/8479</a>.</sub>

**Comment by wilsonge** [comment]
@izharaazmi Can you rebase this please?

**Comment by izharaazmi** [comment]
@wilsonge Done. Please verify.

File: libraries/joomla/utilities/arrayhelper.php [hunk 1]
```diff
@@ -494,29 +423,12 @@ protected static function _sortObjects(&$a, &$b)
 	 *
 	 * @return  array
 	 *
-	 * @see     https://secure.php.net/manual/en/function.array-unique.php
+	 * @see     http://php.net/manual/en/function.array-unique.php
```
Comment (zero-24) [inline libraries/joomla/utilities/arrayhelper.php]: This should stay to be https or do we have any reason to set it to http?

Comment (izharaazmi) [inline libraries/joomla/utilities/arrayhelper.php]: No reason. At the time of actual commit it was so and is retained in rebase.

**Comment by brianteeman** [comment]
Setting back to RTC as it was just a rebase @wilsonge <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8479">issues.joomla.org/joomla-cms/8479</a>.</sub>

**Pull request merged**